File: frontend/src/api/wizard.ts
// 设置向导 API。

import { apiGet, request } from '@/api/client'

export interface WizardState {
  current_step: number
  completed: boolean
  semester_id: number | null
  total_steps: number
  has_semesters: boolean
}

export interface SemesterSummary {
  subjects: number
  teachers: number
  classes: number
  rooms: number
}

export const getWizardState = () => apiGet<WizardState>('/wizard/state')
export const updateWizardState = (body: {
  current_step?: number
  completed?: boolean
  semester_id?: number | null
}) => request<WizardState>('PATCH', '/wizard/state', body)
export const resetWizard = () => request<WizardState>('POST', '/wizard/reset')
export const getSemesterSummary = (id: number) =>
  apiGet<SemesterSummary>(`/semesters/${id}/summary`)

File: backend/app/schemas/wizard.py
"""设置向导 schema。"""

from pydantic import BaseModel


class WizardStateOut(BaseModel):
    current_step: int
    completed: bool
    paused: bool
    semester_id: int | None
    total_steps: int
    has_semesters: bool  # 系统是否已有任何学期(辅助前端判断是否需引导)


class WizardStateUpdate(BaseModel):
    current_step: int | None = None
    completed: bool | None = None
    paused: bool | None = None
    semester_id: int | None = None


class SemesterSummary(BaseModel):
    subjects: int
    teachers: int
    classes: int
    rooms: int
